src/options.c

```diff
@@ -1537,6 +1591,14 @@ void deinit(void)
 	wget_vector_free(&config.accept_patterns);
 	wget_vector_free(&config.reject_patterns);
 
+	for (int i = 0; i < wget_vector_size(config.headers); i++) {
+		wget_http_header_param_t *p = wget_vector_get(config.headers, i);
+		xfree(p->name);
+		xfree(p->value);
+	}
+
```

Comment:
I was already planning on doing that. I'd mentioned it in the TODO in the last commit message. 

Anyhow, I've pushed a new commit with all the above changes. It's ready for merge if you're okay